Documentation improvements (#115)
validatedpatterns/layered-zero-trust: README.md

Showcases the Zero Trust capabilities across Red Hat's product portfolio in a reproducible manner.

### Architecture Overview

The pattern is structured in layers, each building on the one below it:

> * **Validated Patterns Framework** — bootstraps the cluster using OpenShift GitOps (ArgoCD) and the Validated Patterns Operator. All subsequent components are deployed and reconciled by ArgoCD.

Reviewer: ```suggestion
* **Validated Patterns Framework** — bootstraps the cluster using OpenShift GitOps (Argo CD) and the Validated Patterns Operator. All subsequent components are deployed and reconciled by Argo CD.
```

* **Layer 0 — Cluster Hardening** — establishes a secure cluster baseline: TLS certificate management, compliance scanning, advanced cluster management, and runtime security enforcement.
* **Layer 1 — Identity & Secrets** — provisions workload identities (SPIFFE/SPIRE), manages secrets (Vault + ESO), and provides user authentication (Keycloak). Optionally includes a private image registry (Quay + NooBaa).

> * **Layer 2 — Workloads & Supply Chain** — deploys the demo application (qtodo) and, optionally, the full secure supply chain pipeline (RHTAS, RHTPA, OpenShift Pipelines).

Author: Corrected.

See `docs/diagrams/` for logical and schematic diagrams of the pattern.

### Components

Components are grouped by their origin and whether they are always deployed or opt-in.

#### Validated Patterns Framework (provided by the VP project)

These components are part of every Validated Pattern and are not specific to ZTVP. Their Helm charts live in the [validatedpatterns](https://github.com/validatedpatterns) organization.

> * [OpenShift GitOps (ArgoCD)](https://docs.redhat.com/en/documentation/red_hat_openshift_gitops)

Reviewer: ```suggestion
* [OpenShift GitOps (Argo CD)](https://docs.redhat.com/en/documentation/red_hat_openshift_gitops)
```

  * Deploys and continuously reconciles all pattern components via GitOps
* [Validated Patterns Operator](https://validatedpatterns.io)
  * Bootstraps the pattern, manages the Hub/Spoke lifecycle, and drives imperative jobs

#### Core ZTVP Components (always deployed)

These components are deployed by default with every ZTVP installation. Components marked with _(externalized chart)_ use Helm charts maintained in the [validatedpatterns](https://github.com/validatedpatterns) organization rather than local charts in this repository.

* OpenShift cluster hardening
  * [Compliance Operator](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html/security_and_compliance/compliance-operator) _(externalized chart)_
    * Continuously scans the cluster against CIS and other security profiles
  * [cert-manager](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html/security_and_compliance/cert-manager-operator-for-red-hat-openshift)
    * Manages TLS certificates for pattern components

> * [Red Hat Advanced Cluster Management (ACM)](https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.14)

Reviewer: ```suggestion
* [Red Hat Advanced Cluster Management (ACM)](https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes)
```

  * Provides runtime security policy enforcement and image vulnerability scanning
* [HashiCorp Vault](https://www.hashicorp.com/en/products/vault)
  * Secure storage of sensitive assets and dynamic secret generation
* [External Secrets Operator (ESO)](https://external-secrets.io)
  * Synchronizes secrets stored in HashiCorp Vault with OpenShift
* [Red Hat Build of Keycloak (RHBK)](https://access.redhat.com/products/red-hat-build-of-keycloak/) _(externalized chart)_
  * Provides user identities and OIDC authentication for pattern components and workloads
* [Red Hat Zero Trust Workload Identity Manager (ZTWIM)](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html/security_and_compliance/zero-trust-workload-identity-manager) _(externalized chart)_
  * Automates provisioning of SPIFFE/SPIRE-based verifiable identities for workloads running within OpenShift

#### Optional ZTVP Components (opt-in)

These components are commented out in `values-hub.yaml` by default. Uncomment the relevant sections to enable them. See [Deploy the pattern](#deploy-the-pattern) for details.

> * [Red Hat Quay](https://docs.redhat.com/en/documentation/red_hat_quay/3.15) _(externalized chart)_

Reviewer: ```suggestion
* [Red Hat Quay](https://docs.redhat.com/en/documentation/red_hat_quay) _(externalized chart)_
```

  * Enables a private OCI image registry within the environment

> * [Multicloud Object Gateway (NooBaa MCG)](https://docs.redhat.com/en/documentation/red_hat_openshift_container_storage/4.8/html/managing_hybrid_and_multicloud_resources/index)

Reviewer: ```suggestion
* [Multicloud Object Gateway (NooBaa MCG)](https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.20/html/managing_hybrid_and_multicloud_resources/about-the-multicloud-object-gateway)
```

  * Provides S3-compatible object storage for Quay and RHTPA
* [Red Hat Trusted Artifact Signer (RHTAS)](https://docs.redhat.com/en/documentation/red_hat_trusted_artifact_signer/1.3)
  * Provides cryptographic signing and verification of software artifacts and container images

> * [Red Hat Trusted Profile Analyzer (RHTPA)](https://docs.redhat.com/en/documentation/red_hat_trusted_profile_analyzer/2.2)

Reviewer: ```suggestion
* [Red Hat Trusted Profile Analyzer (RHTPA)](https://docs.redhat.com/en/documentation/red_hat_trusted_profile_analyzer)
```
